builtInFunctions/blockDataHandler.go

```diff
@@ -0,0 +1,39 @@
+package builtInFunctions
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type blockDataHandler struct {
+	handler vmcommon.BlockDataHandler
+}
+
+// NewBlockDataHandler returns the block data handler built-in function component
+func NewBlockDataHandler() *blockDataHandler {
+	return &blockDataHandler{}
+}
+
+// SetBlockDataHandler is called when block data handler is set
+func (e *blockDataHandler) SetBlockDataHandler(blockDataHandler vmcommon.BlockDataHandler) error {
+	if check.IfNil(blockDataHandler) {
+		return ErrNilBlockDataHandler
+	}
+
+	e.handler = blockDataHandler
+	return nil
+}
+
+// CurrentRound returns the current round
+func (e *blockDataHandler) CurrentRound() (uint64, error) {
+	if check.IfNil(e.handler) {
```

Comment:
remove this and use the disabled variant on the constructor? Otherwise the checks on L19-L21 are useless